### PR TITLE
Update openshift role to fetch creds from --creds-path

### PR DESCRIPTION
**Pull request by samvarankashyap**
Fixes: #675 

Updates the openshift role to read from both creds path and inline credentials 

openshift credential format is as follows:
```yaml
test:
  api_endpoint: profileexample.com:8443/
  api_token: thisisfromprofile
default:
  api_endpoint: profileexample.com:8443/
  api_token: thisisfromprofile
```

Update topology:
```yaml
---
topology_name: "dummy"
resource_groups:
  - resource_group_name: simple
    resource_group_type: openshift
    resource_definitions:
      - name: openshift
        role: openshift_inline
        data:
          - apiVersion: v1
            kind: ReplicationController
            metadata:
              name: jenkins-slave
              namespace: central-ci-test-ghelling
            spec:
              replicas: 1
              selector:
                name: jenkins-slave
              template:
                metadata:
                  labels:
                    name: jenkins-slave
                spec:
                  containers:
                    - image: redhatqecinch/jenkins_slave:latest
                      name: jenkins-slave
                      env:
                        - name: JENKINS_MASTER_URL
                          value: http://10.8.172.6/
                        - name: JSLAVE_NAME
                          value: mynode
                  restartPolicy: Always
                  securityPolicy:
                    runAsUser: 1000090000
    credentials:
      filename: openshift.yaml
      profile: test
      # inline creds follow the same format
      #api_endpoint: example.com:8443/
      #api_token: mytokentextrighthere
```

### CODE REVIEW

**Comment by samvarankashyap** [comment]
@14rcole done ! :+1:

**Comment by 14rcole** [comment]
Great, thanks!

File: docs/source/examples/workspace/topologies/openshift-new.yml [hunk 1]
```diff
@@ -35,3 +35,6 @@ resource_groups:
     credentials:
       api_endpoint: example.com:8443/
       api_token: mytokentextrighthere
```
Comment (herlo) [inline docs/source/examples/workspace/topologies/openshift-new.yml]: I think we should remove this as the default example. In fact, I don't see any benefit in having api_endpoint/api_token in the PinFile/topology. Essentially, if those PinFiles were ever shared, the api_token becomes compromised.

File: docs/source/openshift.rst [hunk 2]
```diff
@@ -37,6 +37,18 @@ Credentials Management
 An openshift topology can have a ``credentials`` section for each
 :term:`resource_group`, which requires the `api_endpoint`, and the `api_token`
 values.
+Further, Openshift also honours --creds-path in linchpin. The credential file
```
Comment (herlo) [inline docs/source/openshift.rst]: nit: `honors` is the US English version. Update for consistency?

File: docs/source/openshift.rst [hunk 3]
```diff
@@ -37,6 +37,18 @@ Credentials Management
...
+
+.. code-block:: yaml
+
+   ---
+   testprofile:
+       api_endpoint: example.com:8443/
+       api_token: mytokentextrighthere
+   default:
+       api_endpoint: testexample.com:8443/
+       api_token: someothertoken
```
Comment (herlo) [inline docs/source/openshift.rst]: Let's only show filename/profile examples.

File: linchpin/provision/roles/openshift/tasks/provision_resource_group.yml [hunk 4]
```diff
@@ -1,14 +1,55 @@
...
+    auth_var: ""
+
+- name: "set default_cred_profile when res_grp[credentials] is undefined"
+  set_fact:
+    cred_profile: 'default'
+  when: res_grp['credentials'] is not defined
+
+- name: "set default_cred_filename when res_grp[credentials] is undefined"
+  set_fact:
+    cred_filename: 'openshift.yaml'
```
Comment (herlo) [inline linchpin/provision/roles/openshift/tasks/provision_resource_group.yml]: :+1:

**Comment by samvarankashyap** [comment]
@herlo Changes made !

File: docs/source/examples/workspaces/openshift/PinFile [hunk 5]
```diff
@@ -37,3 +37,6 @@ openshift-new:
     credentials:
       api_endpoint: example.com:8443/
       api_token: mytokentextrighthere
+      # alternative way when --creds-path is used 
+      #filename: openshift.yaml
+      #profile: test
```
Comment (herlo) [inline docs/source/examples/workspaces/openshift/PinFile]: This doesn't look to have the new way listed. Is this intentional? I figured moving the main use case to the proper usage would be desired.

Comment (samvarankashyap) [reply docs/source/examples/workspaces/openshift/PinFile]: @herlo I might have just changed the documentation and forgot the workspaces. 
I updated in latest commit .

File: docs/source/examples/workspace/topologies/openshift-new.yml (hunk 1 above)
Comment (herlo) [reply docs/source/examples/workspace/topologies/openshift-new.yml]: @samvarankashyap looks like this one needs to be updated still. Probably squash the commits together and to a force push so they all go in that latest commit.

**Comment by samvarankashyap** [comment]
@herlo done!

**Comment by samvarankashyap** [comment]
@herlo 
bump !

**Comment by samvarankashyap** [comment]
@herlo : A card was added in Jira to add test to openshift . 
I will create a issue on github for the same.

**Comment by herlo** [comment]
@samvarankashyap bump

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**